core/src/test/kotlin/com/segment/analytics/kotlin/core/compat/BuildersTest.kt

```diff
@@ -0,0 +1,224 @@
+package com.segment.analytics.kotlin.core.compat
+
+import kotlinx.serialization.json.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class BuildersTest {
```

Comment:
do u think its better to write these tests in java? or is it the same to have it in kotlin?

Reply:
yeah. that's a good callout. I actually thought about it. the test cases here are mainly for unit tests. so the function invocations are the same in both java and kotlin. I use the way we do it in kotlin as the `expected` result, and the way we do it in java as the `actual` result, to verify they are equivalent. I'm thinking adding integration test cases in `jvm-example` to verify the actual usage in java, so we keep all the code in `core` purely kotlin.